src/bugreporter/odemis_bugreporter.py

```diff
@@ -0,0 +1,542 @@
+            files.append(delphi_calib_reps[-1])
+
+        # Save hw status (if available)
+        ret_code = subprocess.call(['odemis-cli', '--check'])
+        if ret_code in (BACKEND_RUNNING, BACKEND_STARTING):
+            try:
+                # subprocess doesn't have timeout argument in python 2.x, so use future instead
+                f = self._executor.submit(subprocess.check_output, ['odemis-cli', '--list-prop', '*'])
+                props = f.result(60)
+                hwfn = "/tmp/odemis-hw-status.txt"
+                with open(hwfn, 'w+') as f:
+                    f.write(props)
+                files.append(hwfn)
+            except:
+                logging.warning("Cannot save hw status.")
+
+        # Compress files
+        with zipfile.ZipFile(self.zip_fn, "w") as archive:
+            for f in files:
+                if os.path.isfile(f):
+                    archive.write(f, compress_type=zipfile.ZIP_DEFLATED)
+                else:
+                    logging.warning("Bugreporter could not find file %s." % f)
+
+    def _set_description(self, name, email, subject, message):
+        """
+        Saves the description parameters for the ticket creation in a txt file, compresses
+        the file and calls self._create_ticket.
+        :arg name, email, summary, description: (String) arguments for corresponding dictionary
+        keys
+        """
+        self._compress_files_f.result()
+        report_description = {'name': name,
+                              'email': email,
+                              'subject': subject,
+                              'message': message}
+        # Create ticket with special id when testing
+        if TEST_SUPPORT_TICKET:
+            report_description['topicId'] = 12
+
+        with open('/tmp/description.txt', 'a') as f:
+            f.write('Name: %s\n' % name)
+            f.write('Email: %s\n' % email)
+            f.write('Summary: %s\n\n' % subject)
+            f.write('Description:\n%s' % message)
+
+        with zipfile.ZipFile(self.zip_fn, "a") as archive:
+            archive.write('/tmp/description.txt', compress_type=zipfile.ZIP_DEFLATED)
+        wx.CallAfter(self.gui.wait_lbl.SetLabel, "Sending report...")
+        try:
+            api_key = self.search_api_key()
+            self.create_ticket(api_key, report_description, [self.zip_fn])
+            wx.CallAfter(self.gui.Destroy)
+        except Exception as e:
+            logging.warning("Osticket upload failed with exception %s" % e)
+            wx.CallAfter(self.gui.open_failed_upload_dlg)
+
+    def send_report(self, name, email, subject, message):
+        """
+        Calls _set_description in a thread.
+        :arg name, email, summary, description: (String) arguments for corresponding dictionary
+        keys
+        """
+        self._executor.submit(self._set_description, name, email, subject, message)
```

Comment:
As a note (but not need to implement it):
Probably you could make use of the future to report the status of the task to the GUI. So `send_report()` would return the future, the GUI would add a callback on the future, and check whether the task went fine (in which case it closes the GUI), or it raised an exception (in which case it shows the `open_failed_upload_dlg`). This would reduce the coupling of `_set_description()` on the GUI. Unfortunately, there is still the `SetLabel()` needed, so it would still call the GUI anyway...